src/Chapter3.hs

```diff
@@ -753,6 +859,17 @@ parametrise data types in places where values can be of any general type.
   maybe-treasure ;)
 -}
 
+newtype Dragon a = Dragon
+    {
+      magic_power :: a
+    }
+
+data Lair a = Lair
+    {
+      dragon :: Dragon a,
+      treasure_chest :: Maybe Bool
```

Comment:
Interesting. What states did you want to represent with `Maybe Bool`.

* `Just True` - this is when the treasure is present?
* `Just False` - 🤔 
* `Nothing` - no treasures?

I think you could use a simpler type here 🙂